Autostart

drewmoseley opened the pull request.

Commit:
Enable auto start for the Mender Google activation agent

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

Commit:
Move python samples into /opt/gcp and add initial startup script.

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

Commit:
Move Mender Google activation scripts into /opt/gcp

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

drewmoseley commented:
This is not yet ready for merge but it is ready to be reviewed.

drewmoseley commented on image/meta-gcp-iot/recipes-gcp/python-docs-samples/files/start-mqtt-example.sh:
```diff
@@ -0,0 +1,15 @@
+       --project_id "${PROJECT_ID}" \
+       --registry_id "${REGISTRY_ID}" \
+       --device_id "${DEVICE_ID}" \
+       --algorithm RS256 \
```
@Kcr19 should this parameter be RSA256 or is RS256 correct?

Kcr19 replied:
RS256 should work

ptone replied:
with the activation agent - I wasn't initially sure if we still wanted the sample. But I guess it is nice to have a step that the user sees data can be published.

We only allow one MQTT connection per device - so we should probably modify the activation agent script to also check for /data/mender/pre-auth true and quit

drewmoseley replied:
@ptone I can remove the sample script if there is no value in it; I wasn't sure if your activation agent was intended to replace it.  Let me know your preference and I'll be sure to update this PR before merging.

drewmoseley commented:
Note that the start up script still needs to be parameterized for GCP cloud project and such.

Commit:
Add script to manually create Mender certificate

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

drewmoseley commented:
@ptone @Kcr19 I've updated this a bit.  Using _only_ the state script to hold of server connections did not work since the Mender certificate only gets generated after that script returns. I added a systemd pre-exec script that manually creates it and is always invoked before the Mender client is actually launched.  That seems to work and now the certificate is created and the mender client sits waiting for the creation of the preauth_true file.

We still need to make the project ID and such configurable.

Commit:
Create a gcp-config file with variables such as PROJECT_ID.

These are then used by the activation agent and sample python scripts.

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

Commit:
Cleanup FILES_ metadata.

Explicitly include the files by name rather than whole directories

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

drewmoseley commented:
@Kcr19 @ptone I have added some logic to make the PROJECT_ID and other GCP variables configurable.  At build time they are stashed in /opt/gcp/etc/gcp-config.sh.  The activation agent and sample python start script pull the data from there.

As far as I can tell, this PR is ready for review and testing through a full build/test cycle.

Commit:
Use 'rm -rf' rather than 'rmdir' to avoid complaints if it does not exist

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

Commit:
Do not copy the server.crt. We removed that from Yocto a few commits ago

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

drewmoseley commented:
I think this one is ready.  Any comments?

Kcr19 commented:
Looks good to me. Do I do PR against this branch "Autostart" for testing
out ?

On Wed, Aug 15, 2018, 3:30 PM Drew Moseley <notifications@github.com> wrote:

> I think this one is ready. Any comments?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kcr19/mender_gcp_ota_demo/pull/10#issuecomment-413309054>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVs9HfKw8ZvhjE6ubMFFEB-5HY9Fcpveks5uRHbTgaJpZM4V3Pqh>
> .
>

drewmoseley commented:
You should not need another PR for testing. If you manually download the build script and modify it to checkout from my fork and branch instead of yours, that should suffice for testing. The alternative is to simply merge this and then fix any issues in subsequent commits.
